scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -73,7 +75,11 @@ def _decode(self, body, encoding):
                 # http://www.gzip.org/zlib/zlib_faq.html#faq38
                 body = zlib.decompress(body, -15)
         if encoding == b'br' and b'br' in ACCEPTED_ENCODINGS:
-            body = brotli.decompress(body)
+            try:
+                body = brotli.decompress(body)
+            except ImportError:
```

Comment:
This exception cannot happen here as far as I know, can it?

Reply:
I think here I have to make changes here and leave rest as it is, and if not installed than give a warning? Will it do the work?

Reply:
Again, please read the instructions in the issue:

> If the user sets br manually on the Accept-Encoding header, I think we should:
> 
> * Log a warning if brotlipy is not installed.
> * Have the middleware cause an error if the server sends the response br-encoded, and brotlipy is not installed.

So you need to raise an error here, it brotli is not installed, and in addition to this you need to log a warning if brotli is not installed but ACCEPTED_ENCODINGS contains 'br' (not sure which is the best place for this).

Reply:
![Screenshot 2021-03-12 201909](https://user-images.githubusercontent.com/56446640/110956037-3e50f180-8370-11eb-9906-00ba1c93588c.png)

I think this much change can handle the issue?

Reply:
As we are already catching ImportError in this file, you can just set a flag there and check it here.

Also, I'm not sure whether logging an error and returning the compressed body or raising an exception instead is a better way to handle this.